concepts/40 Angular Components/40 Common Features/10 Tree Shaking.md

```diff
@@ -10,4 +12,76 @@ If tree shaking is not configured in your application, import exports from speci
     <!--TypeScript-->
     import { DxButtonModule } from 'devextreme-angular/ui/button';
 
-[tags] angular
\ No newline at end of file
+[note]The tree shaking mechanism works only when you create the **production** bundle.
+
+## Bundle Size
+
+DevExtreme code consists of three main code groups:
+- Utilities Code
+    An application imports Utilities Code, when you add a DevExtreme component. Utilities Code is mandatory for DevExtreme components and the tree shaking mechanim does not remove it. 
+
+- Data Visualization Components Code
+    The bundle includes Data Visualization components code if you use the Chart component in your application. Data Visualization components code has a large size and may affect the resulting bundle.
```

Comment:
1. In both sentences, please add an article before "Data Visualization components code".

2. I'd like to clarify "large size". For example, do you mean "large file size"?